valhalla/midgard/openlr.h

```diff
@@ -108,6 +108,44 @@ struct LocationReferencePoint {
         fow(static_cast<LocationReferencePoint::FormOfWay>(attribute1 & 0x07)) {
   }
 
+  /**
+   * Useful for generating an openlr record from non-openlr data. You can create LRPs with this
+   * constructor and then feed them to the special constructor for the LineLocation
+   * @param longitude
+   * @param latitude
+   * @param bearing
+   * @param frc
+   * @param fow
+   * @param distance
+   * @param lfrcnp
+   */
+  LocationReferencePoint(double longitude,
+                         double latitude,
+                         float bearing,
+                         unsigned char frc,
+                         FormOfWay fow,
+                         LocationReferencePoint* prev,
+                         float distance = 0.f,
+                         unsigned char lfrcnp = 0)
+      : longitude(!prev ? integer2decimal(decimal2integer(longitude))
+                        : prev->longitude +
+                              (std::round((longitude - prev->longitude) * 100000) / 10000)),
+        latitude(!prev ? integer2decimal(decimal2integer(latitude))
+                       : prev->latitude + (std::round((latitude - prev->latitude) * 100000) / 10000)),
```

Comment:
special constructor for LRPs to initialize one from raw data. The other trick here is that since lrps that arent the first one in a linereference are offset in terms of their lat lon we have to do some truncating of precision to make sure they come in and go out the same way. in the end everything is integerized.